### PR TITLE
Move `declare_args` sections from `data_model` into AppConfig.h

### PR DESCRIPTION
**Pull request by andy31415**
Moved these arguments to a central place rather than placing them into the data_model.gni 

Placement in data_model gni has several drawbacks
  - declare_args become nested and harder to find
  - they were passed as compiler command-line arguments, so they are harder to integrate in alternate build systems
  - we have multiple data_model invocations, so having multiple declare_args seems potentially odd scoping (they are likely not scoped, however this is not obvious given the data_model invocation with a name.

### CODE REVIEW

File: src/app/BUILD.gn [hunk 1]
```diff
@@ -43,6 +43,15 @@ declare_args() {
 
   enable_eventlist_attribute = false
 
+  # Allow building ota-requestor-app with a non-spec-compliant floor
+  # (i.e. smaller than 2 minutes) for action delays.
+  non_spec_compliant_ota_action_delay_floor = -1
+
+  # enable type sync client for use in `time-synchronization-server` (if linked)
```
Comment (bzbarsky-apple) [inline src/app/BUILD.gn:50]:
```suggestion
  # enable time sync client for use in `time-synchronization-server` (if linked)
```

File: src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp
Comment (bzbarsky-apple) [inline src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp:1]: This file needs to include the relevant config header now that this is not using a command-line compiler arg, no?

File: src/app/clusters/time-synchronization-server/time-synchronization-server.cpp [hunk 2]
```diff
@@ -37,10 +34,12 @@
 #include <platform/CHIPDeviceLayer.h>
 #include <platform/RuntimeOptionsProvider.h>
 
-#include <app-common/zap-generated/cluster-enums.h>
-
 #include <system/SystemClock.h>
 
+#if TIME_SYNC_ENABLE_TSC_FEATURE
```
Comment (bzbarsky-apple) [inline src/app/clusters/time-synchronization-server/time-synchronization-server.cpp:39]: Is this including the relevant config header?  I don't see it.

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/time-synchronization-server/time-synchronization-server.cpp (hunk 2 above)
Comment (andy31415) [reply src/app/clusters/time-synchronization-server/time-synchronization-server.cpp:39]: the define is used in time-synchronization-server which includes the right include. You want to include it again?

File: src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp
Comment (andy31415) [reply src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp:1]: I missed this one. Added it back.

File: src/app/clusters/time-synchronization-server/time-synchronization-server.cpp (hunk 2 above)
Comment (bzbarsky-apple) [reply src/app/clusters/time-synchronization-server/time-synchronization-server.cpp:39]: If time-synchronization-server.h is including it, probably ok.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yunhanw-google — APPROVED** [review]

**Pull request merged**